testsetup/testsetup.go

```diff
@@ -155,3 +156,26 @@ func EnvLookupFlavorCommand() (flav govim.Flavor, cmd Command, err error) {
 	}
 	return flav, cmd, nil
 }
+
+// RealVimPath iterates over PATH returning the second vim
+// executable found. Reason being that testscript prepends
+// the PATH with the subcommand dir causing a conflict where
+// the first vim found is the subcommand.
+func RealVimPath() (path string, err error) {
+	var firstFound bool
+	for _, dir := range filepath.SplitList(os.Getenv("PATH")) {
+		path := filepath.Join(dir, "vim")
+		d, err := os.Stat(path)
+		if err != nil {
+			continue
+		}
+		if m := d.Mode(); m.IsDir() || m&0111 == 0 {
```

Comment:
Strictly speaking we only need the execute bit set.

Reply:
True. I hope you don't mind me leaving it as is, let me know if so and I'll push another PR.

The rationale is that directories are created with execution bit set by default, so if we (sometime in the future) have a directory named "vim" in the path it will cause a subtle bug that could be tricky to find.